### ISSUE TITLE
<StorageImage /> always displays alt text and loading failure status first!

### ISSUE THREAD

**Issue by xunziheng** [aws-amplify/amplify-ui#6254]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Storage

### How is your app built?

webpack

### What browsers are you seeing the problem on?

Chrome

### Which region are you seeing the problem in?

EU

### Please describe your bug.

I am using StorageImage component in my project like this `<StorageImage alt='logo' path={logoImageKey} />`

When I load this page, I will see alt text and broken image first，then see the truly image I have setup.

I observed the data in the network and found that when the image request has not yet returned, the alt text is displayed instead of loading or blank. that is weird.

<img width="243" alt="image" src="https://github.com/user-attachments/assets/b6620e6a-bc80-4d6d-9a9b-da40f73c25cb" />

### What's the expected behaviour?

show loading or blank instead of alt before image request return.

### Help us reproduce the bug!

`<StorageImage alt='logo' path={logoImageKey} />`

set network to slow 3G

### Code Snippet

```typescript
// Put your code below this line.
<StorageImage alt='logo' path={logoImageKey} />
```

**Comment by cwomack** [comment]
Hello, @xunziheng and thanks for opening this issue.  We are going to mark this as a bug and will follow up with any additional questions or updates we have from here.

**Comment by aadimch** [comment]
Hey, thank you for opening the issue. This issue has been fixed with #6448. Thank you.
